services/autoscaling/src/simcore_service_autoscaling/modules/docker.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from typing import cast
+
+import aiodocker
+from fastapi import FastAPI
+from tenacity._asyncio import AsyncRetrying
+from tenacity.before_sleep import before_sleep_log
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_random_exponential
+
+logger = logging.getLogger(__name__)
+
+
+class AutoscalingDocker(aiodocker.Docker):
```

Comment:
THOUGHT: The "footprint" of a client SDK creation and initialization is very similar: a client object, a ``ping`` function to check responsiveness, an ``on_startup`` to construct, check responsiveness and save one instance in ``app.state`` and a ``on_cleanup`` to graceful shutdown the client.

The api-service has a base class implementation that is already pretty genera byt we can improve it .e.g addin some options `a la tenacity: https://github.com/ITISFoundation/osparc-simcore/blob/e3bfc7315c3eb26d39f875fe2a1a3ccc1a6366ea/services/api-server/src/simcore_service_api_server/utils/client_base.py